src/Utils.php

```diff
@@ -126,4 +126,35 @@ public static function calculateTrailingWhitespaceIndent(Token $token)
 
         return ltrim($str, "\n");
     }
+
+    /**
+     * Sorts the given values using a Schwartzian transform and a custom
+     * comparison function. Order of equal elements is maintained.
+     *
+     * @param array    $elements
+     * @param callable $getComparedValue a callable that takes a single element and returns the value to compare
+     * @param callable $compareValues    a callable that compares two values
+     *
+     * @return array
+     */
+    public static function stableSort(array $elements, callable $getComparedValue, callable $compareValues)
+    {
+        array_walk($elements, function (&$element, $index) use ($getComparedValue) {
+            $element = [$element, $index, $getComparedValue($element)];
+        });
+
+        usort($elements, function ($a, $b) use (&$compareValues) {
+            $comparison = $compareValues($a[2], $b[2]);
+
+            if (0 !== $comparison) {
+                return $comparison;
+            }
+
+            return self::cmpInt($a[1], $b[1]);
```

Comment:
Yes, the method only supports integer keys for now. It's internal and we only pass it arrays with integer keys only.